### ISSUE TITLE
ENH: Add flag for `tensor.nlinalg.eig` to return complex part of eigenvalues 

### ISSUE THREAD

**Issue by jessegrabowski** [pymc-devs/pytensor#418]
### Before

```python
# Currently the complex part of eigenvalues is discarded by `tensor.linalg.eig`:

x = np.array([[-0.2470853 , -0.92681485],
              [ 0.85821264,  1.24789551]])
v, w = pt.linalg.eig(x)
w.eval()
 
#  ComplexWarning: Casting complex values to real discards the imaginary part
#  w[0], v[0] = (z.astype(x.dtype) for z in self._numop(x))
```

### After

```python
v, (w_real, w_imag) = pt.linalg.eig(x, return_complex_part=True)
```

### Context for the issue:

I'm working on some problems that require analysis of modulus of eigenvalues, which I can't do now since the imaginary component is discarded. I know it's not possible to work directly with imaginary numbers in `pytensor`, but it would be easy enough to pass back the imaginary components to users in this case, who could then use them. 

The default would remain unchanged, to prevent legacy code from breaking

**Comment by jessegrabowski** [comment]
Yes, I think I mean gradients. I was remembering [this discourse thread](https://discourse.pymc.io/t/using-pymc3-in-a-model-with-complex-variables-inside/5036/9), and omitted the gradients part from my mind.

So why does `eig` not just return the complex values now?

**Comment by ricardoV94** [comment]
Don't know. I'm fine with your changes but would maybe make it a flag to return complex type TensorVariables? The real and imaginary parts can then be split by the user?

**Comment by jessegrabowski** [comment]
Then the only other (potential) advantage of splitting would be that the gradients for each part could potentially be handled separately?

**Comment by ricardoV94** [comment]
Also useful to see what jax/numba do
